RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -1,32 +1,124 @@
 package com.rfs.common;
 
+import java.time.Duration;
+import java.util.Base64;
+import java.util.List;
+
+import io.netty.buffer.Unpooled;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.lucene.document.Document;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import reactor.util.retry.Retry;
 
 
 public class DocumentReindexer {
     private static final Logger logger = LogManager.getLogger(DocumentReindexer.class);
+    private static final int MAX_BATCH_SIZE = 1000; // Arbitrarily chosen
+
+    public static void reindex(String indexName, Flux<Document> documentStream, ConnectionDetails targetConnection) throws Exception {
```

Comment:
Question from above - why should this take a Flux in?  What would be lost/what would the impact be if you took in a stream and adapted it within this method?